fixtures/tests/_fixtures/test_monkeypatch.py

```diff
@@ -72,14 +108,318 @@ def test_delete_missing_attribute(self):
+            self.assertTrue(1, arg)
+        self.assertEqual(oldmethod, C.foo)
+
+    def test_patch_unboundmethod_with_boundmethod(self):
+        oldmethod = C.foo
+        d = D()
+        fixture = MonkeyPatch(
+            'fixtures.tests._fixtures.test_monkeypatch.C.foo',
+            d.bar_two_args)
+        with fixture:
+            c = C()
+            slf, tgtslf = c.foo()
+            self.expectThat(slf, Is(d))
+            self.expectThat(tgtslf, Is(c))
+        self.assertEqual(oldmethod, C.foo)
+
+    def test_double_patch_instancemethod(self):
+        oldmethod = C.foo
+        oldmethod_inst = C().foo
+        fixture1 = MonkeyPatch(
+            'fixtures.tests._fixtures.test_monkeypatch.C.foo',
+            fake)
+        fixture2 = MonkeyPatch(
+            'fixtures.tests._fixtures.test_monkeypatch.C.foo',
+            fake2)
+        with fixture1:
+            with fixture2:
+                C().foo()
+        self.assertEqual(oldmethod, C.foo)
+        # The method address changes with each instantiation of C, and method
+        # equivalence just tests that. Compare the code objects instead.
+        self.assertEqual(oldmethod_inst.__code__, C().foo.__code__)
+
+    def test_double_patch_staticmethod(self):
+        oldmethod = C.foo_static
+        oldmethod_inst = C().foo_static
+        fixture1 = MonkeyPatch(
+            'fixtures.tests._fixtures.test_monkeypatch.C.foo_static',
+            fake_no_args)
+        fixture2 = MonkeyPatch(
+            'fixtures.tests._fixtures.test_monkeypatch.C.foo_static',
+            fake_static)
+        with fixture1:
+            with fixture2:
+                C.foo_static()
+                C().foo_static()
+        self._check_restored_static_or_class_method(oldmethod, oldmethod_inst,
+            C, 'foo_static')
 
+    def test_double_patch_classmethod(self):
+        oldmethod = C.foo_cls
+        oldmethod_inst = C().foo_cls
+        fixture1 = MonkeyPatch(
+            'fixtures.tests._fixtures.test_monkeypatch.C.foo_cls',
+            fake)
+        fixture2 = MonkeyPatch(
+            'fixtures.tests._fixtures.test_monkeypatch.C.foo_cls',
+            fake2)
+        with fixture1:
+            with fixture2:
+                C.foo_cls()
+                C().foo_cls()
+        self._check_restored_static_or_class_method(oldmethod, oldmethod_inst,
+            C, 'foo_cls')
+
+    def test_patch_c_foo_with_instance_d_bar_self_referential(self):
+        oldmethod = C.foo
+        oldmethod_inst = C().foo
+        fixture = MonkeyPatch(
+            'fixtures.tests._fixtures.test_monkeypatch.C.foo',
+            D().bar_self_referential)
+        with fixture:
+            C().foo()
+        self.assertEqual(oldmethod, C.foo)
+        # The method address changes with each instantiation of C, and method
+        # equivalence just tests that. Compare the code objects instead.
+        self.assertEqual(oldmethod_inst.__code__, C().foo.__code__)
```

Comment:
Given that this assertion happens at least twice, is it worth making a custom Matcher or assertion method? That way you could probably get rid of the comment as well.